README.md

```diff
@@ -700,6 +701,39 @@ interface Table extends Parent {
 
 **Table** represents 2d data.
 
+### CustomCodeComponent
+
+```ts
+interface CustomCodeComponent extends Parent {
+  type: "custom-code-component"
+  /** Repository for the code of the component in the format "[github org]/[github repo]/[component name]". */
+  path: string
+  /** Semantic version of the code of the component, e.g. "^0.3.5". */
+  versionRange: string
+  /** Last date-time where the attributes for this block were modified, in ISO-8601 format. */
+  attributesLastModified: string
+  /** A unique identifier for this instance */
+  id: string
+  /** How the component should be presented in the article page according to the column layout system */
+  layoutWidth:
+      | 'inline'
+	  | 'mid-grid'
+	  | 'full-grid'
+	  | 'full-bleed'
+  /** Configuration data to be passed to the component. */
```

Comment:
Seeing as we use this layoutWidth property in a few places would it make sense for us to define it as an abstract type and use it in place of `string` in the other nodes?

( https://github.com/Financial-Times/content-tree/blob/main/README.md#abstract-types )

something like:

```ts
type LayoutWidth =
      | "auto"
      | "inline"
		| "inset-left"
		| "inset-right"
		| "full-bleed"
		| "full-grid"
```

Reply:
i think it's a good idea!

i'm kind of reticent to add it to all the others rn because i can't find an exhaustive list of layouts so i'd want to go through and test everything properly to make sure i didn't break anything. But i've added it as a type and added it to the customcodecomponent for now, and maybe the other changes can be treated as a different piece of work? Wdyt